compiler/src/iree/compiler/Codegen/LLVMCPU/Passes.cpp

```diff
@@ -52,6 +52,12 @@ static llvm::cl::opt<bool> clEnableMicrokernels(
     llvm::cl::desc("Enables microkernel lowering for vmvx (experimental)"),
     llvm::cl::init(false));
 
+static llvm::cl::opt<bool> clEnableReductionReorderingForAll(
+    "iree-llvmcpu-enable-reduction-reordering-for-all",
```

Comment:
Would suggest being explicit about `fp` in the flag.

Reply:
So, the idea is to keep it generic so that if we support more types (like fixed points, complex, etc), we will still rely on this flag to allow reduction reordering for operations on those types.

Reply:
I would stick to YAGNI and change the flag in the future if at all needed for fixed point or complex but if you like the existing one, we should at least specify what `for-all` means (all codegens, all types, etc.?).

Even better, we could align the flag with MLIR and LLVM:
```
mlir-opt --help | grep fp
        --reassociate-fp-reductions                      - Allows llvm to reassociate floating-point reductions for speed
```
Anyways, I'm bikeshedding here. Don't have a strong opinion. I'll leave it to you.